pkg/controller/gitrepo/gitrepo_reconcile.go

```diff
@@ -137,9 +137,11 @@ func (r *ReconcileGitRepo) Reconcile(request reconcile.Request) (reconcile.Resul
 			reqLogger.Info("keys differed from CRD, keys re-applied to repository")
 		}
 
-		err = r.client.Update(context.TODO(), instance)
-		if err != nil {
-			return err
+		if helpers.AddTenantLabel(&instance.ObjectMeta, instance.Spec.TenantRef.Name) {
+			err := r.client.Update(context.TODO(), instance)
```

Comment:
Now we'll only update a gitRepo if the labels changed. There could be other changes as well that need to be written back...